### PR TITLE
Answer:13

### PR DESCRIPTION
**Pull request by balaji-balamurugan**
style challenge solution - removed all the inputs, added host context to style dynamically.

breaking changes: none

### CODE REVIEW

File: apps/styling/src/app/text.component.ts [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+        color: var(--text-error);
+        font-size: 30px;
+      }
+
+      :host-context(.warning) {
+        color: var(--text-warning);
+        font-size: 25px;
+      }
+    `,
+  ],
```
Comment (tomalaforge) [inline apps/styling/src/app/text.component.ts:32]: error and warning should be inside staticTextComponent. This component should be very generic.

.blue and .font-s-15 are way too specific. If you add new component, you will need to change this component again.

Your css variables are great but make use of them but styling your component from the outside.

Define all properties that can be customize with a default value (you can use a global file as your did, that's fine), but style from the outside.

**Comment by balaji-balamurugan** [comment]
As you mentioned I change the style to staticTextComponent,
But I am not quite sure about styling of this text component, I don't know, I did correctly
If it is wrong, please guide me

**Comment by balaji-balamurugan** [comment]
Yeah @tomalaforge Got it now, thanks for explaining ☺️

**Comment by balaji-balamurugan** [comment]
Hi @tomalaforge, is that fine now?

**Comment by tomalaforge** [comment]
Sorry, but that's not it. 😅
Maybe my exercice is too simple that you can trick it as you did. Because your solution is working. But if my `TextComponent` was a lot more complex, this won't work. 
The goal is to set Css var inside `TextComponent` on each property you want to customize from the outside. 
And on your main component, you can override these Css var with your custom value. 
Is it clearer ?

**Pull request closed without merging**